site/docs/guides/evaling-with-harmbench.md

```diff
@@ -0,0 +1,107 @@
+---
+sidebar_label: Evaluating LLM safety with HarmBench
+---
+
+# Evaluating LLM safety with HarmBench
+
+Recent research has shown that even the most advanced LLMs [remain vulnerable](https://unit42.paloaltonetworks.com/jailbreaking-deepseek-three-techniques/) to adversarial attacks. Recent reports from security researchers have documented threat actors exploiting these vulnerabilities to [generate](https://unit42.paloaltonetworks.com/using-llms-obfuscate-malicious-javascript/) [malware](https://www.proofpoint.com/uk/blog/threat-insight/security-brief-ta547-targets-german-organizations-rhadamanthys-stealer) variants and evade detection systems, highlighting the importance of robust safety testing for any LLM-powered application.
+
+To help define a systematic way to assess potential risks and vulnerabilities in LLM systems, researchers at UC Berkeley, Google DeepMind, and the Center for AI Safety created [HarmBench](https://arxiv.org/abs/2402.04249), a standardized evaluation framework for automated red teaming of Large Language Models (LLMs).
+
+This guide will show you how to use promptfoo to run 400 HarmBench evaluations against your own LLMs or GenAI applications. Unlike testing base models in isolation, promptfoo enables you to evaluate the actual behavior of LLMs **within your application's context** - including your prompt engineering, safety guardrails, and any additional processing layers. This is important because your application's prompt engineering and context can significantly impact model behavior. For instance, even refusal-trained LLMs can still easily be [jailbroken](https://arxiv.org/abs/2410.13886) when operating as an agent in a web browser.
+
+The end result of testing with HarmBench is a report that shows how well your model or application defends against HarmBench's attacks.
+
+![harmbench evaluation results](/img/docs/harmbench-results.png)
+
+## Configure the evaluation
+
+Create a new configuration file `promptfooconfig.yaml`:
+
+```yaml
+# yaml-language-server: $schema=https://promptfoo.dev/config-schema.json
+description: RedTeam evaluation of OpenAI GPT-4o-mini
+targets:
+  - id: openai:gpt-4o-mini
+    label: OpenAI GPT-4o-mini
+prompts:
+  - '{{prompt}}'
+redteam:
+  plugins:
+    - id: harmbench # Tests for harmful content using the HarmBench dataset
+      numTests: 400
+```
+
+## Run the evaluation
+
+In the same folder where you defined `promptfooconfig.yaml`, execute the HarmBench evaluation.
+
+```bash
+npx promptfoo@latest redteam run
+```
+
+Once you're done, view the results:
+
+```bash
+npx promptfoo@latest view
+```
```

Comment:
consider re-adding the harmbench results image here, or screenshotting some other interesting detail in the results

Reply:
Yup, even added some comparative analysis and linked the eval.